builtin/credential/okta/path_groups.go

```diff
@@ -45,34 +47,59 @@ func pathGroups(b *backend) *framework.Path {
 	}
 }
 
-func (b *backend) Group(s logical.Storage, n string) (*GroupEntry, error) {
+// We look up groups in a case-insensitive manner since Okta is case-preserving
+// but case-insensitive for comparisons
+func (b *backend) Group(s logical.Storage, n string) (*GroupEntry, string, error) {
+	canonicalName := n
 	entry, err := s.Get("group/" + n)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	if entry == nil {
-		return nil, nil
+		entries, err := s.List("group/")
+		if err != nil {
+			return nil, "", err
+		}
+		for _, groupName := range entries {
+			if strings.ToLower(groupName) == strings.ToLower(n) {
```

Comment:
Can't we just look at `group/+ strings.ToLower(n)` and avoid looking at all the entries?

Reply:
That will only get us one possible capitalization. So if you have written "GroupFoo" before and now try to write "Groupfoo", you won't find it. With the current logic, you will.

Reply:
Ah, got it!